### PR TITLE
Update with methods example

### PR DESCRIPTION
**Pull request by af-md**

### CODE REVIEW

File: docs/docs/04-core-concepts/01-components.md [hunk 1]
```diff
@@ -57,12 +57,14 @@ func main() {
 This code is unsafe! In code-only components, you're responsible for escaping the HTML content yourself, e.g. with the `templ.EscapeString` function.
 :::
 
-## Methods with components
+## Component as method on types
 
-templ supports methods on types.
+templ Components can be generated as methods on types
```
Comment (a-h) [inline docs/docs/04-core-concepts/01-components.md:62]:
```suggestion
templ components can be returned from methods (functions attached to types).
```

File: docs/docs/04-core-concepts/01-components.md [hunk 2]
```diff
@@ -57,12 +57,14 @@ func main() {
 This code is unsafe! In code-only components, you're responsible for escaping the HTML content yourself, e.g. with the `templ.EscapeString` function.
 :::
 
-## Methods with components
+## Component as method on types
```
Comment (a-h) [inline docs/docs/04-core-concepts/01-components.md:60]:
```suggestion
## Method components
```

**Review by a-h — CHANGES_REQUESTED** [review]
Looks good. You OK a couple of minor changes?

**Review by a-h — APPROVED** [review]

**Comment by af-md** [comment]
@a-h I think we should close this issue: https://github.com/a-h/templ/issues/174  (and assign it to me before closing it)

**Pull request merged**